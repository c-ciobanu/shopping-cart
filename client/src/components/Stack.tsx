import { styled } from "stitches.config";

const Stack = styled("ul", {
	display: "grid",
	shoppingListstyleType: "none",
	margin: 0,
	padding: 0,

	variants: {
		spacing: {
			none: {
				rowGap: 0
			},
			small: {
				rowGap: "$small"
			},
			medium: {
				rowGap: "$medium"
			},
			large: {
				rowGap: "$large"
			}
		}
	}
});

Stack.defaultProps = {
	spacing: "none"
};

export default Stack;
