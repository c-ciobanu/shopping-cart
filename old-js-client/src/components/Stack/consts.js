export const sizes = {
	none: "0px",
	xs: "5px",
	sm: "10px",
	md: "15px",
	base: "20px",
	lg: "30px",
	xl: "35px",
	xxl: "40px"
};
