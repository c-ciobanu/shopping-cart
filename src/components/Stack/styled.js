import styled from "styled-components";

export const StyledDiv = styled.div`
	display: grid;
	grid-row-gap: ${({ gap }) => gap};
`;
