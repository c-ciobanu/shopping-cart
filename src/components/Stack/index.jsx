import React from "react";
import PropTypes from "prop-types";

import { StyledDiv } from "./styled";
import { sizes } from "./consts";

const sizeNames = Object.keys(sizes);

function Stack({ space, children }) {
	return <StyledDiv gap={sizes[space]}>{children}</StyledDiv>;
}

Stack.propTypes = {
	space: PropTypes.oneOf(sizeNames).isRequired,
	children: PropTypes.node.isRequired
};

export default Stack;
