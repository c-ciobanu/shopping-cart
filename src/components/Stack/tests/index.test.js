import React from "react";
import { render } from "@testing-library/react";

import Stack from "../index";

describe("<Stack />", () => {
	it("Expect to not log errors in console", () => {
		const spy = jest.spyOn(global.console, "error");
		render(
			<Stack space="base">
				<p>1</p>
				<p>2</p>
			</Stack>
		);
		expect(spy).not.toHaveBeenCalled();
	});

	it("Should render and match the snapshot", () => {
		const {
			container: { firstChild }
		} = render(
			<Stack space="base">
				<p>1</p>
				<p>2</p>
			</Stack>
		);
		expect(firstChild).toMatchSnapshot();
	});
});
