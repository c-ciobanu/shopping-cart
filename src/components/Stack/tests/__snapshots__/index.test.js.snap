// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stack /> Should render and match the snapshot 1`] = `
<div
  class="sc-AykKC cPBwBZ"
>
  <p>
    1
  </p>
  <p>
    2
  </p>
</div>
`;
